network/default.go

```diff
@@ -823,7 +823,7 @@ func (n *network) processCtrlChan(listener tunnel.Listener) {
 					log.Tracef("Network metric for router %s and gateway %s: %v", event.Route.Router, event.Route.Gateway, metric)
 
 					// check we don't overflow max int 64
-					if d := route.Metric + metric; d > math.MaxInt64 || d <= 0 {
```

Comment:
Can you please make sure that we check that `d` is not `0` i.e. `d <= 0` ? If `route.Metric` and `metric` are `0` then `d` is `0` and we want to handle that properly.

Reply:
My mistake for merging without checking this. Can someone fix.

Reply:
first no value of type int64 is greater than math.MaxInt64. 
second, if we can make sure route.Metric >= 0 and metric >=0.
```
v1 := math.MaxInt64
v2 := math.MaxInt64
fmt.Println(v1+v2) // print -2  
```

Reply:
Yes I get the first part; but as I said I think `route.Metric` and `metric` can be `0` which will yield `d` to be `0` and we want to handle when that happens.

Reply:
Just to elaborate:
If the metric is not set by link (i.e. it's `0`), we can't make any assumption/guarantees about the "quality" of the route hence we set it to the MAX VALUE, because that link might be broken or for whatever reason we can't measure its properly.

Reply:
you are right. thanks for review. i will remove it. sorry for my pool english.  :laughing: 

but it's not intuitive,  if d is zero set to MaxInt64.

Reply:
Yes, it's not very intuitive. In `micro` toolkit we check for it and render infinity character https://github.com/micro/micro/blob/0becdfac18d514a84ce67472d464edcff864a374/network/web/web.go#L232-L233

Reply:
I got it, thanks for clarify.